init commit

utnapischtim opened the pull request.

utnapischtim committed:
init commit

utnapischtim committed:
fix some bugs

philippgualdi commented on invenio_records_dublin_core/config.py:
```diff
@@ -0,0 +1,48 @@
+
+
+# TODO: use invenio-i18n gettext functionality
+def _(x):
```
No translations in the configuration

philippgualdi commented on invenio_records_dublin_core/services/components/original.py:
```diff
@@ -0,0 +1,25 @@
+
+    def create(self, identity, data=None, record=None, errors=None, **kwargs):
+        """Inject parsed view to the record."""
+        print(f"OriginalComponent.create data: {data}, record: {record}")
```
Is this print intended?

utnapischtim replied:
no its not. it will be removed :)

philippgualdi replied:
Ok.

utnapischtim committed:
schema: change to plural attributes

utnapischtim committed:
global: add further facets

utnapischtim committed:
serializer: add access_status

utnapischtim committed:
search: implement result item

rekt-hard commented on invenio_records_dublin_core/assets/semantic-ui/js/invenio_records_dublin_core/search/components/DublinCoreRecordResultsGridItem.js:
```diff
@@ -0,0 +1,30 @@
+  return (
+    <Card fluid key={index} href={`/dublin-core/${result.pid}`}>
+      <Card.Content>
+        <Card.Header>{result.metadata.json.title_statement.title}</Card.Header>
```
Can the metadata from above be used here?

rekt-hard commented on invenio_records_dublin_core/assets/semantic-ui/js/invenio_records_dublin_core/search/components/DublinCoreRecordResultsGridItem.js:
```diff
@@ -0,0 +1,30 @@
+import { Card } from "semantic-ui-react";
+
+export const DublinCoreRecordResultsGridItem = ({ result, index }) => {
+  const metadata = get(result, ["ui", "metadata", "json"], []);
```
Should the default value be a map instead of a list?

rekt-hard commented on invenio_records_dublin_core/assets/semantic-ui/js/invenio_records_dublin_core/search/components/DublinCoreRecordResultsListItem.js:
```diff
@@ -0,0 +1,109 @@
+  const version = get(result, "revision_id", null);
+
+  const publishedAt = undefined;
+  const publicationDate = get(result, "metadata.dates", [
```
What is the difference between these two?

rekt-hard commented on invenio_records_dublin_core/assets/semantic-ui/js/invenio_records_dublin_core/search/components/DublinCoreRecordResultsListItem.js:
```diff
@@ -0,0 +1,109 @@
+
+  const schema = get(result, "original.schema_l10n");
+
+  const [error, setError] = useState("");
```
To be persistent across invenio modules, I would not start introducing the usage of react hooks.
The state is usually set explicitly. E.g.
```js
setState({
  errorMessage: "Descriptive error message"
})
```

rekt-hard commented on invenio_records_dublin_core/resources/serializers/schema.py:
```diff
@@ -0,0 +1,58 @@
+
+    access_status = fields.Function(access_status)
+
+    original = fields.Nested(OriginalSchema, attribute="original")
```
Naming of the variable can be more verbose. `original_record_schema` or something similar would establish a clear context.

rekt-hard commented on invenio_records_dublin_core/resources/serializers/schema.py:
```diff
@@ -0,0 +1,58 @@
+
+    access_status = fields.Function(access_status)
+
+    original = fields.Nested(OriginalSchema, attribute="original")
```
It should probably be moved to the `global-search` module, as it is not part of the dc schema.

utnapischtim replied:
how can i move that functionality to `global-search`?

rekt-hard commented on invenio_records_dublin_core/config.py:
```diff
@@ -0,0 +1,76 @@
+
+DUBLIN_CORE_BASE_TEMPLATE = "invenio_records_dublin_core/base.html"
+
+DUBLIN_CORE_ORIGINAL_SCHEMAS = {}
```
Move to `global-search`

utnapischtim replied:
that is configured in `invenio-global-search` [here](https://github.com/utnapischtim/invenio-global-search/blob/ec9ddd4cbf2d75114c535191458d9c12c349e67e/invenio_global_search/config.py#L12) and could be overwritten in the invenio.cfg file, but the variable is used [here](https://github.com/tu-graz-library/invenio-records-dublin-core/blob/ff71242bf181c580f1ee4c549d21ccaea6e04f8c/invenio_records_dublin_core/resources/serializers/schema.py#L38). i am not sure if i can keep the dublin-core data-model that clean as the other data-models are. dublin-core will have a little bit more functionality because of the usage as the global-search data-model

rekt-hard reviewed (commented):
Nice work.
I think the data models should be implemented as per their specific standard and definition. We might be mixing functionality of a data model module with that of the global search. They should be separated and provide configuration possibilities, such that it can be configured appropriately at an higher level (i.e. `app-rdm or `invenio.cfg`)

utnapischtim committed:
facets: add value_labels to rights

utnapischtim committed:
ui: remove unused code

utnapischtim committed:
global: rename package to invenio-records-global-search

utnapischtim committed:
fix: const not editable

utnapischtim committed:
layout: remove view button

* the view link is in the title an extra button is not necessary

utnapischtim committed:
cli: add configurable rebuild db

utnapischtim committed:
fix: title could have length 2 e.g. language

utnapischtim committed:
schema: improve output

utnapischtim committed:
service: implement custom gs pid

* this pid is necessary to update existing records. the path over pids
  was more complicated, this is a little bit hacky but more straight
  forward.

* the gs_pid is easy to compute. it is the pid from the original with
  the gs- prefix. so it is easy to check if the pid aleady exists and if
  yes to update the record.

utnapischtim committed:
permissions: temporary change

utnapischtim committed:
ui: add collapsable facets

utnapischtim committed:
ui: add global search empty page

* the main reason is to remove the pro tips for rdm-records search

utnapischtim committed:
ui: remove rdm toggle component

* no noticable difference to the default
